### ISSUE TITLE
Compatibility Issue with Docker Compose on Apple Silicon Macs

### ISSUE THREAD

**Issue by Geogouz** [rucio/rucio#6636]
I encountered a problem when trying to set up the Rucio development environment using Docker Compose on an Apple Silicon Mac. For instance, the command ```docker-compose --file etc/docker/dev/docker-compose.yml --profile storage --profile monitoring up -d``` fails because Docker attempts to fetch images for the ARM architecture by default. This occurs despite the Docker Desktop configuration being set up for Linux container virtualization, which typically involves emulating x86-64 architecture on Apple Silicon.

Using the directive ```platform: linux/amd64``` for each image definition (to force the use of AMD64) resolves this compatibility issue. This approach seems to align with the images available on Docker Hub, suggesting that the official support is primarily targeting the x86-64 architecture. Right?

**Comment by rdimaio** [comment]
There is some guidance on this in the docs: https://rucio.github.io/documentation/developer/setting_up_mac_apple_silicon (In particular, [this section](https://rucio.github.io/documentation/developer/setting_up_mac_apple_silicon) talks about the problem you encountered), but I do agree that it would be ideal if we could make the Docker Compose setup just work out of the box on ARM-based machines too, without having to change environment variables.

**Comment by Geogouz** [comment]
Hi, when using the directive `platform: linux/amd64` for `dev_rucio_1` on my Apple Silicon, I noticed that Apache was repeatedly interrupted by _Illegal instruction (4)_ errors, related to the `WSGI` daemon. There is likely an emulation layer's inability to translate these instructions correctly.

I compiled an ARM-based [`rucio-dev`](https://hub.docker.com/r/dxen/rucio-dev) image and indeed things seem to work with it: 

```
rucio:
    container_name: dev_rucio_1
    image: docker.io/dxen/rucio-dev:latest-arm64
    platform: linux/arm64/v8
```

Since I didn’t see any other ARM version of `rucio-dev` on the hub, I was wondering whether you have come across other Mac cases that solved this with a different workaround (or didn't face the same problem at all).

**Comment by NadiaSteyn** [comment]
Hi @Geogouz - your comment helped me to run Rucio on ARM64, thanks. One comment: the image tag "latest-arm64" gave me the following error:
```
rucio Error
Error response from daemon: manifest for dxen/rucio-dev:latest-arm64 not found: manifest unknown: manifest unknown

```
But it works when I remove the tag,  i.e. (in the docker-compose.yml file):
```
rucio:
    container_name: dev_rucio_1
    image: docker.io/dxen/rucio-dev
    platform: linux/arm64/v8
```

And I also had to change the "rucioclient" section:
```
  rucioclient:
    image: docker.io/dxen/rucio-dev
    platform: linux/amd64
```

**Comment by Geogouz** [comment]
Hi @NadiaSteyn. I recently started playing a bit with the auto-building of the involved images (for arm64) and pushing of those to my hub space (https://hub.docker.com/repositories/dxen). Since the Git workflow I used is based on the official one, the names/tags generated are also going to be like the official ones. This is why the "latest-arm64 not found" I guess. 

However, keep in mind that **this is by no means an actually official release** (I have I guess to include this in the repo description). It is more like a personal testing project as there are several conventions I am following for some images to successfully have them working for aarch64 architecture. I am mostly doing it to see what works and what fails.. even though things seem to work so far for me (I think :D).

I will update my repo branch today with my setup and share it here in case you find it useful.
